File: src/ELibrary.Backend/ShopApi/Features/StatisticsFeature/Domain/Dtos/GetBookStatisticsRequest.cs
namespace ShopApi.Features.StatisticsFeature.Domain.Dtos
{
    public class GetBookStatisticsRequest
    {
        public DateTime? FromUTC { get; set; }
        public DateTime? ToUTC { get; set; }
        public StatisticsBook[] IncludeBooks { get; set; }
    }
}

File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/shop/statistics/getBookStatisticsRequest.ts
import { GetBookStatistics, mapBookToStatisticsBook, StatisticsBook } from "../../../..";

export interface GetBookStatisticsRequest {
    fromUTC: Date | null,
    toUTC: Date | null,
    includeBooks: StatisticsBook[]
}

export function mapGetBookStatisticsToGetBookStatisticsRequest(statistics: GetBookStatistics): GetBookStatisticsRequest {
    return {
        fromUTC: statistics.fromUTC,
        toUTC: statistics.toUTC,
        includeBooks: statistics.includeBooks.map(x => mapBookToStatisticsBook(x))
    }
}